mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -35,8 +35,15 @@ import org.apache.spark.rdd.RDD
  * @param predictionAndLabels an RDD of (predicted ranking, ground truth set) pairs.
  */
 @Since("1.2.0")
```

Comment:
Oh, we need to update the `@Since` tags. I think you can somehow add a `@Since` annotation to the default constructor args? as it's since 3.4.0. I'm not sure exactly where it goes. The old constructor can remain since 1.2.0. If that doesn't work, maybe we can leave this constructor and add the new one as a new `def this(...)` since 3.4.0?

Reply:
I made changes and updated the docs.

Reply:
I think that we should refer to `BinaryClassificationMetrics` and `MulticlassMetrics`, in which `RDD[_ <: Product]` was used as the input.

Reply:
since end users now mainly use the `.ml`, is there any plan to expose this function to `.ml`?

Reply:
Sorry in advance for my poor understanding. I have some questions.
- What is `.ml` ? Do you mean `org.apache.spark.ml` ?
- `RDD[_ <: Product]` is used to make`MulticlassMetric` class available to `.ml` ?

Reply:
For the first one - yeah this change is in the 'older' .mllib package. I don't think there is an equivalent for it in the DataFrame-based .ml packages, so, maybe we can ignore that here. But if nDCG is supported in the .ml package somewhere and I forgot it, would be good to add it there too.

The declaration suggested here might actually work for both input types without a separate constructor. Try it maybe? if it works, yes, that is simpler, and lets this API support even more inputs

Reply:
Thank you for your explanation!

Could you review this?
https://github.com/apache/spark/pull/36920